src/main/python/lambda_handler.py

```diff
@@ -4,108 +4,118 @@
+                             aws_session_token=credentials['SessionToken'])
+        topic_arns = list_sns(sns)
+        print(len(topic_arns))
+
+    # Sync Source DDB to Destination Region
+    CLIENT_NAME = 'dynamodb'
+    source_ddb = session.client(CLIENT_NAME, region_name=source_region, aws_access_key_id=credentials['AccessKeyId'],
+                                aws_secret_access_key=credentials['SecretAccessKey'],
+                                aws_session_token=credentials['SessionToken'])
+
+    destination_ddb = session.client(
+        CLIENT_NAME, region_name=destination_region)
+    sync_ddb_table(source_ddb, destination_ddb)
+
+
+# Scan returns paginated results, so only partial data will be copied
+def sync_ddb_table(source_ddb, destination_ddb):
+    response = source_ddb.scan(
+        TableName="table1"
+    )
+    for item in response['Items']:
+        destination_ddb.put_item(
+            TableName="table2",
+            Item=item
+        )
+
+# This code uses a mutable default argument and modifies it to return. This would leak results across calls
+
+
+def list_sns(sns, topics=[]):
+    response = sns.list_topics()
+    for topic_arn in response["Topics"]:
+        topics.append(topic_arn["TopicArn"])
+    return topics
+
+
+# Infinite loop because a list is modified while being iterated over, Indices are not updated.
+def infinite_loop():
+    words = ['aws', 'amazon', 'codeguru']
+    for w in words:
+        if len(w) > 4:
+            words.insert(0, w)
+    return words
+
+# Prefer DefaultDict over setDefult
+
+
+def setdefault_example():
+    std_dict = dict()
+    for k, v in enumerate(range(5)):
+        std_dict.setdefault(k, []).append(v)
+    return std_dict
+
+# This method reads multiple file paths, open each file to load data, but forgets to close it, leading to resource leaks
+# Further, it selectively processes content based on string find condition. The find() operation can be simply replaced
+# with a membership tests because one does not need to know the position at which the search keyword appears.
+
+
+def process_kinesis(kinesis, file_list_path):
+    flp = open(file_list_path, 'r')
+    for line in flp:
+        file_path = line.strip('\n').strip('\r\n')
+        fp = open(file_path, 'r')
+        for content in fp:
+            if content.find("kinesis") != -1:
+                record = load_kinesis_record(content)
+                save_kinesis_record(kinesis, record)
+
+# Do not call this function unless you're sure that the "cmd" is secure to run
+# This function can be misused to carry out shell injection attacks.
+# Further, the code is simply passing an exception, which is not the best practice
+# Further, the code keeps track of successful loads by writing to a global variable; which can lead to inaccuracies in
+# case of concurrent read/writes to the global variable.
+
+
+def load_kinesis_record(cmd, mode='subprocess'):
+    global successes
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Global variables are dangerous because they can be simultaneously accessed from multiple sections of a program. This frequently results in bugs. Most bugs involving global variables arise from one function reading and acting on the value of a global variable before another function has the chance to set it to an appropriate value. We recommend using a configuration module to mutate global state.

[Learn more](https://docs.python.org/3/faq/programming.html#how-do-i-share-global-variables-across-modules)